osquery/tables/system/python_packages.cpp

```diff
@@ -113,28 +136,120 @@ void genSiteDirectories(const std::string& site,
     r["directory"] = site;
     r["path"] = directory;
     r["pid_with_namespace"] = "0";
+    r["uid"] = BIGINT(user_id);
     results.push_back(r);
   }
 }
 
-void genWinPythonPackages(const std::string& keyGlob,
-                          QueryData& results,
-                          Logger& logger) {
+std::vector<std::string> listWinPythonPaths(const std::string& keyGlob) {
 #ifdef WIN32
+  std::vector<std::string> results;
+
   std::set<std::string> installPathKeys;
-  expandRegistryGlobs(keyGlob, installPathKeys);
+  auto status = expandRegistryGlobs(keyGlob, installPathKeys);
+  if (!status.ok()) {
+    return {};
+  }
+
   QueryData pythonInstallLocation;
   for (const auto& installKey : installPathKeys) {
     queryKey(installKey, pythonInstallLocation);
     for (const auto& p : pythonInstallLocation) {
       if (p.at("name") != "(Default)") {
         continue;
       }
-      genSiteDirectories(p.at("data"), results, logger);
+      results.push_back(p.at("data"));
     }
     pythonInstallLocation.clear();
   }
+  return results;
 #endif
+  return {};
+}
+
+std::vector<std::string> traverseVersions(const std::string& path) {
+  std::vector<std::string> versions;
+  if (!listDirectoriesInDirectory(path, versions, false).ok()) {
+    return {};
+  }
+
+  std::vector<std::string> all_paths;
+  for (const auto& version : versions) {
+    auto version_path = fs::path(version);
+    if (fs::is_symlink(symlink_status(version_path))) {
+      continue;
+    }
+
+    auto path = version + "/lib/python%/site-packages";
+    std::vector<std::string> sites;
+    resolveFilePattern(path, sites);
+    for (const auto& site : sites) {
+      all_paths.push_back(site);
+    }
+  }
+
+  return all_paths;
+}
+
+std::vector<std::map<std::string, UserPath>> getUserPathList(
+    const QueryContext& context) {
+  std::vector<std::map<std::string, UserPath>> paths_list;
+
+  auto user_list = usersFromContext(context);
```

Comment:
On Windows, if you just do `SELECT * FROM python_packages;` (without the `uid` constraint), do you get the full list of users?

Reply:
I was incorrect in my original statement

On my branch:
```
runas /user:Administrator "powershell C:\Users\konst\osquery\build\osquery\Debug\osqueryi.exe --verbose"
```
<img width="1161" alt="Screenshot 2025-01-10 at 12 13 22 PM" src="https://github.com/user-attachments/assets/d49cd136-56d5-40e5-aebd-26ff5eed3267" />

On master:
```
runas /user:Administrator "powershell C:\Users\konst\osquery\build\osquery\Debug\osqueryi.exe --verbose"
```
<img width="1184" alt="Screenshot 2025-01-10 at 12 18 07 PM" src="https://github.com/user-attachments/assets/43733fef-bded-4104-9e7a-feb04edfc29d" />

Which means I need to delete that comment I added in `specs/python_packages.table`. I believe this breaks original behavior. 🤦